### ISSUE TITLE
Unable to complete deployment with awx-operator on FC34 and minikube v1.23.0

### ISSUE THREAD

**Issue by p4rancesc0** [ansible/tower-operator#533]
##### ISSUE TYPE
 - awx-operator pod logs looping forever, not completing installation

##### SUMMARY
can't connect to awx service as it doesn't seems available.
If connecting via bash to the awx operator pod awx doesn't seems to be installed.

##### ENVIRONMENT
* AWX version: X.Y.Z
* Operator version: 0.13.0 0.12.0 devel
* Kubernetes version: v1.22.1
* AWX install method: minikube v1.23.0 with docker on linux

##### STEPS TO REPRODUCE

install docker rootless
install minikube: [https://minikube.sigs.k8s.io/docs/](https://minikube.sigs.k8s.io/docs/)
follow [https://github.com/ansible/awx-operator#basic-install](https://github.com/ansible/awx-operator#basic-install)

so: 

`[awxuser@awxfc ~]$ minikube delete
* Deleting "minikube" in docker ...
* Deleting container "minikube" ...
* Removing /home/awxuser/.minikube/machines/minikube ...
* Removed all traces of the "minikube" cluster.`

`[awxuser@awxfc ~]$ minikube start --addons=ingress --cpus=6 --cni=flannel --install-addons=true     --kubernetes-version=stable --memory=12g  --container-runtime=containerd
* minikube v1.23.0 on Fedora 34 (kvm/amd64)
* Automatically selected the docker driver. Other choices: none, ssh
! Your cgroup does not allow setting memory.
  - More information: https://docs.docker.com/engine/install/linux-postinstall/#your-kernel-does-not-support-cgroup-swap-limit-capabilities
! docker is currently using the fuse-overlayfs storage driver, consider switching to overlay2 for better performance
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Creating docker container (CPUs=6, Memory=12288MB) ...
* Preparing Kubernetes v1.22.1 on containerd 1.4.9 ...
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Configuring Flannel (Container Networking Interface) ...
* Verifying Kubernetes components...
  - Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.0
  - Using image gcr.io/k8s-minikube/storage-provisioner:v5
  - Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.0
  - Using image k8s.gcr.io/ingress-nginx/controller:v1.0.0-beta.3
* Verifying ingress addon...
* Enabled addons: storage-provisioner, default-storageclass, ingress
* kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default`

Now create a pvc, pod was stuck in "1 pod has unbound immediate persistentvolumeclaims" otherwise

`[awxuser@awxfc ~]$ kubectl apply -f awx-storage.yml
persistentvolumeclaim/postgres-awxfc-postgres-0 created`

`[awxuser@awxfc ~]$ cat awx-storage.yml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: postgres-awxfc-postgres-0
spec:
  storageClassName: standard
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi`  

now back to the site procedure:

`[awxuser@awxfc ~]$ kubectl apply -f https://raw.githubusercontent.com/ansible/awx-operator/0.13.0/deploy/awx-operator.yaml
customresourcedefinition.apiextensions.k8s.io/awxs.awx.ansible.com created
customresourcedefinition.apiextensions.k8s.io/awxbackups.awx.ansible.com created
customresourcedefinition.apiextensions.k8s.io/awxrestores.awx.ansible.com created
clusterrole.rbac.authorization.k8s.io/awx-operator created
clusterrolebinding.rbac.authorization.k8s.io/awx-operator created
serviceaccount/awx-operator created
deployment.apps/awx-operator created
`

`[awxuser@awxfc ~]$ kubectl apply -f awx-fc.yml
awx.awx.ansible.com/awxfc created
`

`[awxuser@awxfc ~]$ cat awx-fc.yml
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
  name: awxfc
spec:
  service_type: nodeport
  ingress_type: none
  hostname: awxfc.fu.priv
`

##### EXPECTED RESULTS

can connect to awx.

##### ACTUAL RESULTS

awx operator pod not showing with this command
`[awxuser@awxfc ~]$ kubectl get pods -l "app.kubernetes.io/managed-by=awx-operator"
NAME               READY   STATUS    RESTARTS   AGE
awxfc-postgres-0   1/1     Running   0          3m48s
`
logs lopping forever

here's the only one I've identified as and error:

`PLAY RECAP *********************************************************************
localhost                  : ok=29   changed=0    unreachable=0    failed=1    skipped=27   rescued=0    ignored=0   
\n","job":"2703501726821866378","name":"awxfc","namespace":"default","error":"exit status 2","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error
	pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/runner.(*runner).Run.func1
	src/github.com/operator-framework/operator-sdk/pkg/ansible/runner/runner.go:239"}
{"level":"error","ts":1631103614.5857651,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"awx-controller","request":"default/awxfc","error":"event runner on failed","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error
	pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil
	pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil
	pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until
	pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:90"}
`

##### ADDITIONAL INFORMATION

`[awxuser@awxfc ~]$ kubectl get events --sort-by='.metadata.creationTimestamp'
LAST SEEN   TYPE     REASON                    OBJECT                                            MESSAGE
8m47s       Normal   NodeHasSufficientPID      node/minikube                                     Node minikube status is now: NodeHasSufficientPID
8m47s       Normal   NodeHasNoDiskPressure     node/minikube                                     Node minikube status is now: NodeHasNoDiskPressure
8m47s       Normal   NodeHasSufficientMemory   node/minikube                                     Node minikube status is now: NodeHasSufficientMemory
8m22s       Normal   NodeHasNoDiskPressure     node/minikube                                     Node minikube status is now: NodeHasNoDiskPressure
8m22s       Normal   NodeHasSufficientMemory   node/minikube                                     Node minikube status is now: NodeHasSufficientMemory
8m22s       Normal   Starting                  node/minikube                                     Starting kubelet.
8m22s       Normal   NodeHasSufficientPID      node/minikube                                     Node minikube status is now: NodeHasSufficientPID
8m22s       Normal   NodeAllocatableEnforced   node/minikube                                     Updated Node Allocatable limit across pods
8m15s       Normal   NodeReady                 node/minikube                                     Node minikube status is now: NodeReady
8m15s       Normal   RegisteredNode            node/minikube                                     Node minikube event: Registered Node minikube in Controller
5m22s       Normal   Provisioning              persistentvolumeclaim/postgres-awxfc-postgres-0   External provisioner is provisioning volume for claim "default/postgres-awxfc-postgres-0"
5m22s       Normal   ExternalProvisioning      persistentvolumeclaim/postgres-awxfc-postgres-0   waiting for a volume to be created, either by external provisioner "k8s.io/minikube-hostpath" or manually created by system administrator
5m22s       Normal   ProvisioningSucceeded     persistentvolumeclaim/postgres-awxfc-postgres-0   Successfully provisioned volume pvc-0d370f0d-b106-41de-be44-ba8e7cf3a9f4
4m34s       Normal   ScalingReplicaSet         deployment/awx-operator                           Scaled up replica set awx-operator-69c646c48f to 1
4m34s       Normal   SuccessfulCreate          replicaset/awx-operator-69c646c48f                Created pod: awx-operator-69c646c48f-8p5qt
4m34s       Normal   Scheduled                 pod/awx-operator-69c646c48f-8p5qt                 Successfully assigned default/awx-operator-69c646c48f-8p5qt to minikube
4m33s       Normal   Pulling                   pod/awx-operator-69c646c48f-8p5qt                 Pulling image "quay.io/ansible/awx-operator:0.13.0"
3m39s       Normal   Created                   pod/awx-operator-69c646c48f-8p5qt                 Created container awx-operator
3m39s       Normal   Pulled                    pod/awx-operator-69c646c48f-8p5qt                 Successfully pulled image "quay.io/ansible/awx-operator:0.13.0" in 53.875775096s
3m39s       Normal   Started                   pod/awx-operator-69c646c48f-8p5qt                 Started container awx-operator
2m28s       Normal   SuccessfulCreate          statefulset/awxfc-postgres                        create Pod awxfc-postgres-0 in StatefulSet awxfc-postgres successful
2m28s       Normal   Scheduled                 pod/awxfc-postgres-0                              Successfully assigned default/awxfc-postgres-0 to minikube
2m27s       Normal   Pulling                   pod/awxfc-postgres-0                              Pulling image "postgres:12"
2m1s        Normal   Started                   pod/awxfc-postgres-0                              Started container postgres
2m1s        Normal   Pulled                    pod/awxfc-postgres-0                              Successfully pulled image "postgres:12" in 25.904976338s
2m1s        Normal   Created                   pod/awxfc-postgres-0                              Created container postgres
`

**Comment by tobgen** [comment]
Yea, more reports that it does not work in 1.22.x.
Something has clearly changed in later versions of k8s that makes the deployment not working anymore.

**Comment by p4rancesc0** [comment]
Thanks, 

please let me know if you are going to fix the issue.

BR
Francesco

**Comment by PotentialIngenuity** [comment]
I have the same setup and problem. Here is another error i see:

`failed: [localhost] (item=service_account) => 
{
  "ansible_loop_var": "item", 
  "changed": false, 
  "item": "service_account", 
  "msg": "Failed to find exact match for rbac.authorization.k8s.io/v1beta1.RoleBinding by [kind, name, singularName, shortNames]"
}`

**Comment by PotentialIngenuity** [comment]
I have dug into this further and found that 1.22.x removed several beta api endpoints, see [link](https://www.p3r.one/kubernetes-1-22-notes/)

Changing v1beta1 to v1 yields the following results. Using the [service_account](https://github.com/ansible/awx-operator/blob/devel/roles/installer/templates/service_account.yaml.j2) template

```
11:40:16 # kubectl apply -f rbac.yml
error: unable to recognize "rbac.yml": no matches for kind "RoleBinding" in version "rbac.authorization.k8s.io/v1beta1"
```
```
11:46:35 # kubectl apply -f rbac.yml
rolebinding.rbac.authorization.k8s.io/ansible-awx created
```

**Comment by ghost** [comment]
Hi,
Same problem here.
Also ried with Minikube 1.20.0 and 1.21.0.
This is the final test with 1.20.0
```
eplicas\": 1, \"updatedReplicas\": 1}}}\n\r\nPLAY RECAP *********************************************************************\r\nlocalhost                  : ok=30   changed=0    unreachable=0    failed=1    skipped=27   rescued=0    ignored=0   \r\n\n","job":"8574153963535421338","name":"awx-demo","namespace":"default","error":"exit status 2","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/runner.(*runner).Run.func1\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/runner/runner.go:239"}
{"level":"error","ts":1631284867.202643,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"awx-controller","request":"default/awx-demo","error":"event runner on failed","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tpkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:90"}
```

```
$ kubectl get pods
NAME                            READY   STATUS    RESTARTS   AGE
awx-demo-7bbb564887-jhc2q       0/4     Pending   0          62m
awx-demo-postgres-0             1/1     Running   0          62m
awx-operator-849b7d7d9d-stxpf   1/1     Running   0          65m
```

**Comment by p4rancesc0** [comment]
Hi, 

I switched back on centos 8.4.

I managed with this:

`minikube start --addons=ingress --cpus=6 --cni=flannel --install-addons=true     --kubernetes-version=stable --memory=12g --kubernetes-version 1.20.2`

`minikube.x86_64                      1.20.0-0 `

Got an error on start, really odd path...

did

```
minikube addons disable ingress
minikube addons enable ingress
```

than start again.

BR
Francesco

**Comment by p4rancesc0** [comment]
The problem really seems to be kubernetes version.

**Comment by gatboccia** [comment]
I am seeing the same issue in GKE Version  1.20.8-gke.2100

**Comment by p4rancesc0** [comment]
I managed with kubernetes 1.20.2 and minikube 1.20.0-0 on Centos 8.4

Br
Francesco

**Comment by gatboccia** [comment]
> I managed with kubernetes 1.20.2 and minikube 1.20.0-0 on Centos 8.4
> 
> Br
> Francesco

So perhaps the api's were changed around >=1.20.3

**Comment by chrismeyersfsu** [comment]
@tchellomello can you look at this?

**Comment by tobgen** [comment]
I have done some more testing, and at least for me, things works in 1.21.x. (Note that I'm not using Minicube)

Did the following, started out with a 1.21.2 cluster, installed the operator and then applied the demo-config.
AWX pods tries to start (will not start because no db is configured nor a PVC is created)

Deleted everything, upgraded to 1.21.3. Rinse and repeat.
AWX pods tries to start (Expected)

Deleted everything, upgraded to 1.21.4, Rinse and repeat.
AWX pods tries to start (Expected)

Deleted everything, upgraded to 1.22.2, Rinse and repeat.
AWX pods never shows up in the list, no events giving any indication that they even try to start.

```NAME    STATUS   ROLES                  AGE     VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
k8s-1   Ready    control-plane,master   6m11s   v1.21.2   192.168.71.5   <none>        Ubuntu 20.04.3 LTS   5.4.0-84-generic   docker://20.10.8
NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
awx           awx-7948bf7847-kqft9               0/4     Pending   0          12s
awx           awx-postgres-0                     0/1     Pending   0          21s
default       awx-operator-69c646c48f-stnhz      1/1     Running   0          2m58s



NAME    STATUS   ROLES                  AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
k8s-1   Ready    control-plane,master   13m   v1.21.3   192.168.71.5   <none>        Ubuntu 20.04.3 LTS   5.4.0-84-generic   docker://20.10.8
NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
awx           awx-7948bf7847-vkdj5               0/4     Pending   0          17s
awx           awx-postgres-0                     0/1     Pending   0          26s
default       awx-operator-69c646c48f-fp74l      1/1     Running   0          66s



NAME    STATUS   ROLES                  AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
k8s-1   Ready    control-plane,master   20m   v1.21.4   192.168.71.5   <none>        Ubuntu 20.04.3 LTS   5.4.0-84-generic   docker://20.10.8
NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
awx           awx-7948bf7847-l67t4               0/4     Pending   0          7s
awx           awx-postgres-0                     0/1     Pending   0          15s



NAME    STATUS   ROLES                  AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
k8s-1   Ready    control-plane,master   27m   v1.22.1   192.168.71.5   <none>        Ubuntu 20.04.3 LTS   5.4.0-84-generic   docker://20.10.8
NAMESPACE     NAME                               READY   STATUS    RESTARTS        AGE
awx           awx-postgres-0                     0/1     Pending   0               105s
default       awx-operator-69c646c48f-gvb72      1/1     Running   0               2m26s

**Comment by tchellomello** [comment]
@chrismeyersfsu  I'll take a look  :+1:

**Comment by ghost** [comment]
As a side note, I am using Debian 10 but I don't think that this really matters in this case.

**Comment by slaurencelle** [comment]
Hello achrjulien,

i can confirm you that it is not your OS causing the problem but more the kubernetes version 1.22. There is around 4 or 5 discussion here about this bug, but still nothing yet from the developper, hope they can solve it soon. Don't want to have to reinstall all my kuberntes cluster because of that :(

Stephane

**Comment by vrubiolo** [comment]
I am seeing the same issue here too (this is also Fedora 34, using Minikube `1.23.1` and K8s `1.22.1`).
I have redeployed w/ K8S 1.21.3 as a workaround and confirms this works (I can reach the AWX homepage via the Minikube service URL)

**Comment by ShaLLx64** [comment]
seems like it wouldn't work for me creating it on K3 under a namespace called awx but it worked when I created it in the default namespace instead.

**Comment by shanemcd** [comment]
Will be fixed in #569

**Comment by shanemcd** [comment]
Closing - #569 has been released in 0.14.0.
